File: spring/src/main/java/com/okaka/onenightjinroh/application/domain/Game.java
package com.okaka.onenightjinroh.application.domain;

public class Game {
    public Long gameId;
    public Room room;
    public Rule rule;

    public Game(Long gameId) {
        this.gameId = gameId;
        this.room = null;
        this.rule = null;
    }

    public Game(Long gameId, Room room, Rule rule) {
        this.gameId = gameId;
        this.room = room;
        this.rule = rule;
    }

    public Long getGameId() {
        return gameId;
    }

    public Room getRoom() {
        return room;
    }

    public Rule getRule() {
        return rule;
    }
}

File: spring/src/main/java/com/okaka/onenightjinroh/application/domain/Game.kt
package com.okaka.onenightjinroh.application.domain

class Game(
    var gameId: Long?,
    var roomId: Long?,
    var rule: Rule?,
    var term: GameTerm? = GameTerm.NIGHT,
) {
    companion object {
        fun createByRoomId(roomId: Long): Game {
            return Game(null, roomId, null, null)
        }

        fun singleCreate(gameId: Long): Game {
            return Game(gameId, null, null, null)
        }

        fun startGame(roomId: Long, participantCount: Int): Game {
            return Game(
                null,
                roomId,
                Rule.createByParticipantCount(participantCount),
                term = GameTerm.NIGHT
            )
        }
    }

    fun changeTerm(term: GameTerm): Game {
        return Game(
            this.gameId,
            this.roomId,
            this.rule,
            term
        )
    }

}
